src/components/node/NodeConfig.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useEffect, useState } from 'react';
+import Input from '../common/Input';
+import Checkbox from '../common/Checkbox';
+import {
+  NodeConfigContainer,
+  NodeConfigTitle,
+  NodeConfigButton,
+  ErrorText,
+  SuccessText
+} from '../../styles/NodeConfigStyles';
+import { UpdateNodeConfigParams, CommandResponse, NodeDetails } from '../../hooks/useNodeManagement';
+
+interface NodeConfigProps {
+  selectedNode: NodeDetails;
+  onConfigUpdate: (config: UpdateNodeConfigParams) => Promise<CommandResponse>;
+  onClose: () => void;
+}
+
+const NodeConfig: React.FC<NodeConfigProps> = ({ selectedNode, onConfigUpdate }) => {
+  const [serverPort, setServerPort] = useState<number>(selectedNode.node_ports.server_port);
+  const [swarmPort, setSwarmPort] = useState<number>(selectedNode.node_ports.swarm_port);
+  const [nodeName, setNodeName] = useState<string>(selectedNode.name);
+  const [runOnStartup, setRunOnStartup] = useState<boolean>(selectedNode.run_on_startup);
+  const [error, setError] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    setServerPort(selectedNode.node_ports.server_port);
+    setSwarmPort(selectedNode.node_ports.swarm_port);
+    setNodeName(selectedNode.name);
+    setRunOnStartup(selectedNode.run_on_startup);
+    setError(null);
+    setSuccessMessage(null);
+  }, [selectedNode]);
+
+  const handleUpdateConfig = async () => {
+    setError(null);
+    setSuccessMessage(null);
+    try {
+      const result = await onConfigUpdate({
+        originalNodeName: selectedNode.name,
+        nodeName: nodeName,
+        serverPort: serverPort,
+        swarmPort: swarmPort,
+        runOnStartup: runOnStartup,
+      });
+      
+      if (result.success) {
```

Comment:
Not sure whats the return type of result. But if it throws error success won't be available and might throw error. Double check this please 
```suggestion
      if (result?.success) {
```